addons/udes_stock/tests/test_stock_picking_user_assignment.py

```diff
@@ -0,0 +1,242 @@
+            with self.subTest(user=user.name):
+                self.assertFalse(user.u_picking_id)
+
+        # Manually assign a user to the picking
+        users.assign_picking_to_users(self.goods_in_picking)
+        for user in users:
+            with self.subTest(user=user.name):
+                self.assertEqual(user.u_picking_id, self.goods_in_picking)
+
+    def test_assign_additional_user_to_picking_with_u_user_process_partial_enabled(self):
+        """
+        Test if u_user_process_partial is enabled, another user can be assigned
+        """
+        self.picking_type_goods_in.u_user_process_partial = True
+        user = self.inbound_user
+        self.assertEqual(self.goods_in_picking.state, "assigned")
+
+        # Assert nothing currently assigned
+        self.assertFalse(user.u_picking_id)
+
+        # Manually assign a user to the picking
+        user.assign_picking_to_users(self.goods_in_picking)
+        self.assertEqual(user.u_picking_id, self.goods_in_picking)
+        user_start_time = user.u_picking_assigned_precise_time
+        picking_start_time = self.goods_in_picking.u_date_started
+
+        # Assign another user to the picking
+        user_2 = self.inbound_user_2
+        user_2.assign_picking_to_users(self.goods_in_picking)
+
+        self.assertEqual(picking_start_time, self.goods_in_picking.u_date_started)
+        self.assertEqual(user.u_picking_id, self.goods_in_picking)
+        self.assertEqual(user.u_picking_assigned_precise_time, user_start_time)
+        self.assertEqual(user_2.u_picking_id, self.goods_in_picking)
+        self.assertGreater(user_2.u_picking_assigned_precise_time, user_start_time)
+
+    def test_try_assign_user_to_already_assigned_picking(self):
+        """
+        Test to check that a user is not re-assigned if they are already assigned to it.
+        """
+        self.assertEqual(self.goods_in_picking.state, "assigned")
+
+        # Manually assign a user to the picking
+        self.inbound_user.assign_picking_to_users(self.goods_in_picking)
+        self.assertEqual(self.inbound_user.u_picking_id, self.goods_in_picking)
+        self.assertTrue(self.inbound_user.u_picking_assigned_precise_time)
+        # Try to assign the same picking
+        start_time = self.inbound_user.u_picking_assigned_precise_time
+        self.inbound_user.assign_picking_to_users(self.goods_in_picking)
+        self.assertEqual(self.inbound_user.u_picking_id, self.goods_in_picking)
+        self.assertEqual(start_time, self.inbound_user.u_picking_assigned_precise_time)
+
+    def test_un_assign_user_with_u_user_process_partial_enabled(self):
+        """Test un assignment of a picking when multiple users can exist on it"""
+        self.picking_type_goods_in.u_user_process_partial = True
+        user = self.inbound_user
+        # Manually assign a user to the picking
+        user.assign_picking_to_users(self.goods_in_picking)
+        self.assertTrue(user.u_picking_id)
```

Comment:
Again, we should assert that the assigned picking is the expected picking, so an `assertEqual` would be better here.